pkg/command/service/common.go

```diff
@@ -146,16 +150,99 @@ func GetIngressController(p *pkg.PerfParams) map[string]string {
 	return ingressController
 }
 
+// GenerateJSONOutput generates CSV file from the rows data
+func GenerateCSVOutput(rows [][]string, outputPathPrefix string) (csvPath string, err error) {
+	csvPath = outputPathPrefix + ".csv"
+	err = utils.GenerateCSVFile(csvPath, rows)
+	if err != nil {
+		fmt.Printf("failed to generate CSV file and skip %s\n", err)
+		return "", err
+	}
+	return csvPath, nil
+}
+
+// GenerateHTMLOutput generates HTML file from CSV file
+func GenerateHTMLOutput(csvPath string, outputPathPrefix string) (htmlPath string, err error) {
+	htmlPath = outputPathPrefix + ".html"
+	err = utils.GenerateHTMLFile(csvPath, htmlPath)
+	if err != nil {
+		fmt.Printf("failed to generate HTML file and skip %s\n", err)
+		return "", err
+	}
+	return htmlPath, nil
+}
+
+// GenerateJSONOutput generates JSON output from the result
+func GenerateJSONOutput(result interface{}, outputPathPrefix string) (jsonPath string, err error) {
+	jsonData, err := json.Marshal(result)
+	if err != nil {
+		fmt.Printf("failed to generate json data and skip %s\n", err)
+		return "", err
+	}
+	jsonPath = outputPathPrefix + ".json"
+	err = utils.GenerateJSONFile(jsonData, jsonPath)
+	if err != nil {
+		fmt.Printf("failed to generate json file and skip %s\n", err)
+		return "", err
+	}
+	return jsonPath, nil
+}
+
+// GenerateOutputPathPrefix generates the prefix of output path, which can be combined with a suffix name(.csv) to form a complete path
+func GenerateOutputPathPrefix(inputsOutput string, outputFilenameFlag string) (pathPrefix string, err error) {
+	current := time.Now()
+	outputLocation, err := utils.CheckOutputLocation(inputsOutput)
+	if err != nil {
+		fmt.Printf("failed to check measure output location: %s\n", err)
+		return "", err
+	}
+	pathPrefix = filepath.Join(outputLocation, fmt.Sprintf("%s_%s", current.Format(DateFormatString), outputFilenameFlag))
+	return pathPrefix, nil
+}
+
+// GenerateOutput generates outputs according to flags(csvFlag, htmlFlag and josnFlag) from rows and result
+func GenerateOutput(inputsOutput string, outputFilenameFlag string, csvFlag bool, htmlFlag bool, jsonFlag bool, rows [][]string, result interface{}) error {
+	outputPathPrefix, err := GenerateOutputPathPrefix(inputsOutput, outputFilenameFlag)
+	if err != nil {
+		return err
+	}
+	if csvFlag && rows != nil {
+		// generate csv file from rows
+		csvPath, err := GenerateCSVOutput(rows, outputPathPrefix)
+		if err == nil {
```

Comment:
What if these `Generate...()` calls failed?

Reply:
Thank you, the err returns of `GenerateCSVOutput` may cause to mistake, I can delete it and just use csvFlag to instead.